sql/rowexec/insert.go

```diff
@@ -30,20 +32,24 @@ import (
 )
 
 type insertIter struct {
-	schema              sql.Schema
-	inserter            sql.RowInserter
-	replacer            sql.RowReplacer
-	updater             sql.RowUpdater
-	rowSource           sql.RowIter
-	lastInsertIdUpdated bool
-	hasAutoAutoIncValue bool
-	ctx                 *sql.Context
-	insertExprs         []sql.Expression
-	updateExprs         []sql.Expression
-	checks              sql.CheckConstraints
-	tableNode           sql.Node
-	closed              bool
-	ignore              bool
+	schema                sql.Schema
+	inserter              sql.RowInserter
+	replacer              sql.RowReplacer
+	updater               sql.RowUpdater
+	rowSource             sql.RowIter
+	lastInsertIdUpdated   bool
+	lastInsertUuidUpdated bool
+	hasAutoAutoIncValue   bool
+	ctx                   *sql.Context
+	insertExprs           []sql.Expression
+	insertTuples          [][]sql.Expression
```

Comment:
I wouldn't have this field or the one below if you can help it. Just use the row being inserted.

Reply:
I don't think we can get the fully correct behavior using just the row of final values, even for `auto_increment`. For example, `last_insert_id()` should record the generated auto_increment ID when the column is explicitly specified in the insert statement and the `DEFAULT` keyword is used to insert the auto_increment value. In Dolt, we do insert the right value, but we don't update the value returned from `last_insert_id()` (Bug https://github.com/dolthub/dolt/issues/7565).

It doesn't seem like we can account for that case with the current model where we just have the insert projection expressions and the final row values. It seems like we need to look at the original values in the insert tuples (for `VALUES` at least) to determine if the value is eligible to be stored as a last insert value.